### DISCUSSION TITLE
Prisma generated types can hinder changing the database layer later on?

### DISCUSSION THREAD

**Discussion by ahmednawaz10p** [prisma/prisma#8191] [Q&A]
I recently heard about `prisma` and wanted to try it. I really liked the simple interface it provides and the level of type saftey. however, I have one question. when we use `prisma` to query an entity, the response we get has a type that comes from `@prisma/client` ( prisma generated ). Though it has some advantages but this is where I think that a violation of open closed principle can occur. In the future if I want to change the database layer or the orm to `typeorm` then all the return types of my service methods have to be changed to reflect my entities (rather than previously prisma generated types). Think of it in context of nestjs where we have entities that are our common return types. Is there a way we can solve this problem with prisma? Or it is not a problem at all?

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
